### PR TITLE
Tikeeva add Sending CV.feature

### PR DESCRIPTION
**Pull request by tikeeva-qa**

### CODE REVIEW

File: qa/automation/CareerPortal.Tests.Scenarios/4. Sending CV/4.1.Sending CV.feature [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        | Ж'еня-Тикеева, мл. | tikeeva1998@mail.ru | cv.rtf 2Mb  | True     |
+	When user clicks "Send" button
+    Then user should see the message that CV was submitted
+
+    Examples: 
+		| name of the page                                               |
+		| https://career.quantori.com/ru                                 |
+		| https://career.quantori.com/ru/positions/cloud-python-engineer |
+
+@Type:Positive
```
Comment (VladTeryaev90) [inline qa/automation/CareerPortal.Tests.Scenarios/4. Sending CV/4.1.Sending CV.feature:36]: Scenario 4.1_02 dublicatates of 4.1_03. I would be choose 4.1_03  but with following edits :
In 39 row after word "values" I would be delete ":"
You missed key word "Examples" before you table of values ( see attach)
Also for max coverage I would be add cv file with size 9,9 Mb to test that our form will be send

File: qa/automation/CareerPortal.Tests.Scenarios/4. Sending CV/4.1.Sending CV.feature [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+@Priority:Critical
+Scenario: 4.1_03 Successful submission with filled "Name", "Email", "Message" fields, checkbox and attached CV
+    Given user entered "<Name>", "<Email>", "<Message>", "<Attach CV>", "<Checkbox>" fields with foolowing combination of values:
+        | Name               | Email               | Message                    | Attach CV   | Checkbox |
+        | Evgeniia T         | tikeeva_es@mail.ru  | Please consider my resume. | cv.pdf 10Mb | True     |
+        | Евгения Т          | tikeeva1998@mail.ru | I will be a good worker.   | cv.doc 2Mb  | True     |
+        | Evgeniia T S       | tikeeva_es@mail.ru  | Please consider my resume. | cv.docx 2Mb | True     |
+        | Ж'еня-Тикеева, мл. | tikeeva1998@mail.ru | I will be a good worker.   | cv.rtf 2Mb  | True     |
+	When user clicks "Send" button
+    Then user should see the message that CV was submitted
```
Comment (VladTeryaev90) [inline qa/automation/CareerPortal.Tests.Scenarios/4. Sending CV/4.1.Sending CV.feature]: Scenario 4.1_02 dublicatates of 4.1_03. I would be choose 4.1_03  but with following edits :
In 39 row after word "values" I would be delete ":"
You missed key word "Example" before you table of values ( see attach)
Also for max coverage I would be add cv file with size 9,9 Mb to test that our form will be send

File: qa/automation/CareerPortal.Tests.Scenarios/4. Sending CV/4.1.Sending CV.feature [hunk 3]
```diff
@@ -0,0 +1,132 @@
+Feature: 4.1. Sending CV
+
```
Comment (VladTeryaev90) [inline qa/automation/CareerPortal.Tests.Scenarios/4. Sending CV/4.1.Sending CV.feature:2]: I would be add "Background" with Given which described that user should opened https://career.quantori.com/ru
And scrolled down to the CV form

File: qa/automation/CareerPortal.Tests.Scenarios/4. Sending CV/4.1.Sending CV.feature [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+    When user attaches a file in "<Extension>" format
+	And file size is less than or equal to 10 Mb
+	Then attached file name should be displayed under the "Attach CV" button
+
+	Examples:
+        |Extension|
+		|.pdf     |
+		|.doc     |
+		|.docx    |
+		|.rtf     |
```
Comment (VladTeryaev90) [inline qa/automation/CareerPortal.Tests.Scenarios/4. Sending CV/4.1.Sending CV.feature]: Dublicate 4.1_03 scenario ( we already have verification for size and format of file "Attach CV")

File: qa/automation/CareerPortal.Tests.Scenarios/4. Sending CV/4.1.Sending CV.feature (hunk 2 above)
Comment (VladTeryaev90) [reply qa/automation/CareerPortal.Tests.Scenarios/4. Sending CV/4.1.Sending CV.feature]: Maybe it is correct rename Scenario to Scenario Outline..

Comment (tikeeva-qa) [reply qa/automation/CareerPortal.Tests.Scenarios/4. Sending CV/4.1.Sending CV.feature]: 1. You're right, the grouping makes sense in this case, thanks. I will fix it.
2. After values you need ":" and don't need "Example", since I'm using a table.
image
3. We have a 10MB boundary condition, so to cover all successful tests we need values less than 10MB (example 2Mb) and 10MB.

Comment (VladTeryaev90) [reply qa/automation/CareerPortal.Tests.Scenarios/4. Sending CV/4.1.Sending CV.feature]: The same for others scenarious where you have values

Comment (VladTeryaev90) [reply qa/automation/CareerPortal.Tests.Scenarios/4. Sending CV/4.1.Sending CV.feature]: Also for best coverege yuu can add value "empty" in "Message" to show that CV form will be send

File: qa/automation/CareerPortal.Tests.Scenarios/4. Sending CV/4.1.Sending CV.feature (hunk 3 above)
Comment (tikeeva-qa) [reply qa/automation/CareerPortal.Tests.Scenarios/4. Sending CV/4.1.Sending CV.feature:2]: Thanks for your comment! 
I'll think about it)

File: qa/automation/CareerPortal.Tests.Scenarios/4. Sending CV/4.1.Sending CV.feature (hunk 4 above)
Comment (tikeeva-qa) [reply qa/automation/CareerPortal.Tests.Scenarios/4. Sending CV/4.1.Sending CV.feature]: Thanks for your comment!) 
I'll fix it)

File: qa/automation/CareerPortal.Tests.Scenarios/4. Sending CV/4.1.Sending CV.feature (hunk 2 above)
Comment (tikeeva-qa) [reply qa/automation/CareerPortal.Tests.Scenarios/4. Sending CV/4.1.Sending CV.feature]: Ok, you're right) I'll fix it)

File: qa/automation/CareerPortal.Tests.Scenarios/4. Sending CV/4.1.Sending CV.feature (hunk 4 above)
Comment (VladTeryaev90) [reply qa/automation/CareerPortal.Tests.Scenarios/4. Sending CV/4.1.Sending CV.feature]: Mistake "foolowing" in 22, 40, 52 rows

**Review by VladTeryaev90 — APPROVED** [review]

File: qa/automation/CareerPortal.Tests.Scenarios/4. Sending CV/4.1.Sending CV.feature [hunk 5]
```diff
@@ -0,0 +1,107 @@
+Feature: 4.1. Sending CV
+
+@Type:Positive
+@Priority:Critical
+Scenario Outline: 4.1_01 Correct display of the cv submit form
+    Given user wanted to send CV
```
Comment (Georgiy-Petrov-Quantori) [inline qa/automation/CareerPortal.Tests.Scenarios/4. Sending CV/4.1.Sending CV.feature:6]: Excess step

File: qa/automation/CareerPortal.Tests.Scenarios/4. Sending CV/4.1.Sending CV.feature [hunk 6]
```diff
@@ -0,0 +1,107 @@
+Feature: 4.1. Sending CV
+
+@Type:Positive
+@Priority:Critical
+Scenario Outline: 4.1_01 Correct display of the cv submit form
```
Comment (Georgiy-Petrov-Quantori) [inline qa/automation/CareerPortal.Tests.Scenarios/4. Sending CV/4.1.Sending CV.feature:5]: Let's just say "Send CV" form

File: qa/automation/CareerPortal.Tests.Scenarios/4. Sending CV/4.1.Sending CV.feature [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+		| name of the page                                               |
+		| https://career.quantori.com/ru                                 |
+		| https://career.quantori.com/ru/positions/cloud-python-engineer |
+
+@Type:Positive
+@Priority:Critical
+Scenario Outline: 4.1_02 Successful submission with filled "Name", "Email" fields, checkbox and attached CV
+    Given user founded the Send CV form on page "<name of the page>"
+    And user entered "<Name>", "<Email>", "<Message>", "<Attach CV>", "<Checkbox>" fields with following combination of values:
+        | Name               | Email               | Message                    | Attach CV   | Checkbox |
```
Comment (Georgiy-Petrov-Quantori) [inline qa/automation/CareerPortal.Tests.Scenarios/4. Sending CV/4.1.Sending CV.feature:23]: Should be moved to examples

File: qa/automation/CareerPortal.Tests.Scenarios/4. Sending CV/4.1.Sending CV.feature [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+    Examples: 
+		| name of the page                                               |
+		| https://career.quantori.com/ru                                 |
+		| https://career.quantori.com/ru/positions/cloud-python-engineer |
+
+@Type:Positive
+@Priority:Critical
+Scenario Outline: 4.1_02 Successful submission with filled "Name", "Email" fields, checkbox and attached CV
+    Given user founded the Send CV form on page "<name of the page>"
+    And user entered "<Name>", "<Email>", "<Message>", "<Attach CV>", "<Checkbox>" fields with following combination of values:
```
Comment (Georgiy-Petrov-Quantori) [inline qa/automation/CareerPortal.Tests.Scenarios/4. Sending CV/4.1.Sending CV.feature:22]: This step should be placed in "When" section

File: qa/automation/CareerPortal.Tests.Scenarios/4. Sending CV/4.1.Sending CV.feature [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+Scenario Outline: 4.1_03 If important fields are empty, no sending occurs, errors are highlighted
+    Given user entered "<Name>", "<Email>", "<Attach CV>", "<Checkbox>" fields with following combination of values:
+        | Name               | Email               | Attach CV   | Checkbox |
+        |                    |                     |             |          |
+        |                    | tikeeva_es@mail.ru  | cv.pdf 10Mb | True     |
+        | Евгения Т          |                     | cv.doc 2Mb  | True     |
+        | Evgeniia T S       | tikeeva_es@mail.ru  |             | True     |
+        | Ж'еня-Тикеева, мл. | tikeeva1998@mail.ru | cv.rtf 2Mb  |          |
+    When user clicks "Send" button
+    Then user should see a red border and error text under empty field, no sending occurs
```
Comment (Georgiy-Petrov-Quantori) [inline qa/automation/CareerPortal.Tests.Scenarios/4. Sending CV/4.1.Sending CV.feature:48]: Need to separate this step into several results

File: qa/automation/CareerPortal.Tests.Scenarios/4. Sending CV/4.1.Sending CV.feature (hunk 7 above)
Comment (Georgiy-Petrov-Quantori) [inline qa/automation/CareerPortal.Tests.Scenarios/4. Sending CV/4.1.Sending CV.feature:23]: What is path to CV? Where from user uploads it (location on computer)?

File: qa/automation/CareerPortal.Tests.Scenarios/4. Sending CV/4.1.Sending CV.feature [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+
+@Type:Negative
+@Priority:Critical
+Scenario Outline: 4.1_04 If the fields are filled in incorrectly, sending does not occur, errors are highlighted
+    Given user enters "<Name>", "<Email>", "<Attach CV>", "<Checkbox>" fields with following combination of values:
+        | Name               | Email               | Attach CV   | Checkbox |
+        | 123456 8102        | tikeeva_es@mail.ru  | cv.pdf 10Mb | True     |
```
Comment (Georgiy-Petrov-Quantori) [inline qa/automation/CareerPortal.Tests.Scenarios/4. Sending CV/4.1.Sending CV.feature:55]: Could we use here pairwise testing?

File: qa/automation/CareerPortal.Tests.Scenarios/4. Sending CV/4.1.Sending CV.feature [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+       | евгения@mail.ru |
+       | test@.          |
+       | @mail.ru        |
+    When user clicks "Send" button
+    Then user should see a red border and text "Invalid email format" under "Email" field, no sending occurs
+
+@Type:Negative
+@Priority:Major
+Scenario Outline: 4.1_06 Validation is triggered for invalid "Name"
+    Given user filled "Name" field with "<Input>" values:
```
Comment (Georgiy-Petrov-Quantori) [inline qa/automation/CareerPortal.Tests.Scenarios/4. Sending CV/4.1.Sending CV.feature:77]: Could be combined with previous case

File: qa/automation/CareerPortal.Tests.Scenarios/4. Sending CV/4.1.Sending CV.feature [hunk 12]
```diff
@@ -0,0 +1,107 @@
...
+       | Evgeniia %  |
+       | Евгения  1  |
+    When user clicks "Send" button
+    Then user should see a red border and text "Invalid name format" under "Name" field, no sending occurs
+
+@Type:Negative
+@Priority:Major
+Scenario: 4.1_07 Validation is triggered for a message that is too long
+    Given user filled the Send CV form
+    When user types 501 or more characters in the "Message" field
```
Comment (Georgiy-Petrov-Quantori) [inline qa/automation/CareerPortal.Tests.Scenarios/4. Sending CV/4.1.Sending CV.feature:92]: What about boundary What about boundary values?

**Review by Georgiy-Petrov-Quantori — CHANGES_REQUESTED** [review]

File: qa/automation/CareerPortal.Tests.Scenarios/4. Sending CV/4.1.Sending CV.feature (hunk 6 above)
Comment (tikeeva-qa) [reply qa/automation/CareerPortal.Tests.Scenarios/4. Sending CV/4.1.Sending CV.feature:5]: Thanks for your comment!
Ok, I'll fix it.

File: qa/automation/CareerPortal.Tests.Scenarios/4. Sending CV/4.1.Sending CV.feature (hunk 5 above)
Comment (tikeeva-qa) [reply qa/automation/CareerPortal.Tests.Scenarios/4. Sending CV/4.1.Sending CV.feature:6]: Ok, I'll fix it, thanks)
